src/index.js

```diff
@@ -1,7 +1,349 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
+import React from "react";
+import ReactDOM from "react-dom";
+import VTMap from "./Components/VTMap.js";
+import Button from "./Components/Buttons.js";
+import Display from "./Components/InfoBox.js";
+import borderData from "./Components/border.js";
+import leafletPip from "leaflet-pip";
+import L from "leaflet";
+import GuessBox from "./Components/guessBox.js"
+import "./CSS/index.css" 
+import Move from "./Components/Move.js"
+import GameWon from "./Components/gameWon.js"
 
-ReactDOM.render(
-  <h1>Hello World!</h1>,
-  document.getElementById('root')
-)
\ No newline at end of file
+let localStore = window.localStorage;
+
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+
+    this.state = {
+      initialLat: 43.9,
+      initialLng: -72.7317,
+      currentLat: 43.9,
+      currentLng: -72.7317,
+      town: "?",
+      county: "?",
+      latDisplay: "?",
+      lngDisplay: "?",
+      startDisabled: false,
+      guessDisabled: true,
+      quitDisabled: true,
+      zoom: 8,
+      modal: false,
+      winModal: false,
+      score: 100,
+      moveArr: [],
+      allScores: localStore.getItem("scores") || "",
+      highScore: "",
+      userName: ""
+    };
+  }
+
+//Getting all scores
+componentDidMount(){
+  let allScores= this.state.allScores;
+  JSON.parse(allScores); 
+  let highestScore= JSON.parse(allScores).reduce((previousValue, nextValue)=>{
+    if (parseInt(previousValue.score) > parseInt (nextValue.score)){
+      return previousValue
+    }else{
+      return nextValue
+    }
+  });
+  console.log(highestScore)
+  this.setState({
+    highScore: highestScore.score + ' ' + highestScore.name
+  })
+}
+
+
+  //When user clicks on start, the start button is disabled and the guess and quit button are enabled
+  handleStart = (evt) => {
+    evt.preventDefault();
+    let start = randomLocation();
+    this.setState({
+      startDisabled: true,
+      guessDisabled: false,
+      quitDisabled: false,
+      currentLat: start[1],
+      currentLng: start[0],
+      initialLat: start[1],
+      initialLng: start[0],
+      moveArr: this.state.moveArr.concat([[start[1], start[0]]]),
+      zoom: 18,
+    });
```

Comment:
You never want to use `this.state` inside of a `this.setState` you can work around this by making your `this.setState` a callback function and let your `moveArr:` equal to `prevState.moveArr`

Or you could break it down and set them to variables and manipulate them outside of the set state then pass the variable through to your new `moveArr`